### ISSUE TITLE
Info for getting PSGet work with personal access token (PAT)

### ISSUE THREAD

**Issue by jianyunt** [MicrosoftDocs/PowerShell-Docs#1200]
Issue Details
-----------------
I have seen  several issues opened related to accessing packages to VSTS feed.

https://github.com/PowerShell/PowerShellGet/issues/59
https://github.com/PowerShell/PowerShellGet/issues/91
https://github.com/OneGet/oneget/issues/196
https://github.com/OneGet/oneget/issues/174

Here are some info that are useful for users to get it to work. So please add them somewhere under PackageManagement or PowerShellGet.
	 
	# Create PAT token to access your VSTS feed
	# Reference: https://www.visualstudio.com/en-us/docs/setup-admin/team-services/use-personal-access-tokens-to-authenticate#create-personal-access-tokens-to-authenticate-access
	 
	# Enter your PAT token as password
	$patCred = Get-Credential john@sampledomain.com
	 
	# Register your VSTS package management feed using your PAT-base token
	Register-PSRepository psgetvsts -SourceLocation https://myrepo.pkgs.visualstudio.com/_packaging/psgettestfeednuget/v2/ -PublishLocation https://myrepo.pkgs.visualstudio.com/_packaging/psgettestfeed/nuget/v2/ -InstallationPolicy Trusted -Credential $patCred -Verbose
	 
	# Find modules from your VSTS Repositoryo
	Find-Module -Repository psgetvsts -Credential $patCred
	 
	# Install modules from your VSTS Repository
	Install-module -Name Contoso -Repository psgetvsts -Credential $patCred -Verbose -Scope CurrentUser
	 
	# Publish modules to your VSTS package management feed using your PAT-base token.	 
	# For publishing, it is required to add your VSTS feed with username and PAT token, so that NuGet.exe uses these credential during push operation.
	NuGet.exe sources Add -Name "psgettestfeed" -Source "https://myrepo.pkgs.visualstudio.com/_packaging/psgettestfeed/nuget/v2" -UserName manikb@microsoft.com -Password $myPatToken
	 
	Publish-Module -Name MyModule -Repository psgetvsts -Credential $patCred -Verbose
	 
	For VSTS feeds, NuGetApiKey is can be any random string, like -NuGetApiKey 'VSTS' .

Version(s) of document impacted
------------------------------
- [x] Impacts 6 document
- [x] Impacts 5.1 document
- [x] Impacts 5.0 document
- [ ] Impacts 4.0 document
- [ ] Impacts 3.0 document

Reason(s) for not selecting all version of documents
----------------------------------------------------
- [ ] The documented feature was introduced in selected version of PowerShell
- [ ] This issue only shows up in selected version of the document

**Comment by jianyunt** [comment]
Keith will coordinate with Michael to review on this content.

**Comment by arcotek-ltd** [comment]
Thank you for this, but why does it need a PAT and how do we use our (Azure) AD credentials instead?

**Comment by arcotek-ltd** [comment]
Ok. [I see](https://docs.microsoft.com/en-us/vsts/package/nuget/nuget-exe?view=vsts), so it's because I am using Nuget v2, which only supports PAT.

I am using v2 because that is what I have read I need to use with PowerShellGet. Does PowerShellGet 1.6.5 now support v3?

Thanks

**Comment by wgv-srsedate** [comment]
Is there a way to enter the credential at PSRepository registration time, and have it not require a credential every time you want to interact with that feed?  I'd hate to have to keep track of a PAT token over a long period of time, or have to re-generate a new token every time I want to install or update packages from my feed.

**Comment by SydneyhSmith** [comment]
@wgv-srsedate credential persistence is one of the top items in our backlog right now and we are hoping to get to it very soon, but currently Register-PSRepository does not have this functionality

**Comment by Heybrajham** [comment]
as @arcotek-ltd said, I can't get it works with V3, I'm surprised that almost a year has passed, and still no support for V3

**Comment by chasewilson** [comment]
Hi all, Thank you for the feedback and information. I'm updating the page in [this PR](https://github.com/MicrosoftDocs/PowerShell-Docs/pull/5226) to reference using Azure DevOps as a PowerShell repository. There is a link to [this how-to guide](https://docs.microsoft.com/en-us/azure/devops/artifacts/tutorials/private-powershell-library?view=azure-devops) which fully explains the use of PAT's and how to connect.

Once the PR is merged, this issue will be closed but if you have any more questions or issues, feel free to reach out or create a new issue. 

Thank you again for all the contribution!
